classifides/ads/views.py

```diff
@@ -1,2 +1,136 @@
-from django.shortcuts import render
-from django.http import HttpResponse
+from django.shortcuts import render,get_object_or_404, redirect
+from .models import *
+from ads.forms import *
+from django.contrib.auth.decorators import login_required
+from django.contrib import messages
+from django.utils.decorators import method_decorator
+from django.views import View
+from django.http import Http404
+
+def home(request):
+    categories = Category.objects.all()
+    return render(request, 'base.html', {'categories': categories})
+
+def category_ads(request, category_id):
+    category = get_object_or_404(Category, id=category_id)
+    ads = Ad.objects.filter(category=category).prefetch_related('images')
+    categories = Category.objects.all()
+    return render(request, 'category.html', {
+        'category': category,
+        'ads': ads,
+        'categories': categories,
+    })
+
+class MyAdsView(View):
+    @method_decorator(login_required)
+    def get(self, request):
+        categories = Category.objects.all()
+        ads = Ad.objects.filter(user=request.user)
+        return render(request, 'my_ads.html', {'ads': ads, 'categories': categories})
+    
+    @method_decorator(login_required)
+    def post(self, request):
+        ad_id = request.POST.get('ad_id')
+        if ad_id:
+            ad = get_object_or_404(Ad, id=ad_id)
+            ad.delete()
+            messages.success(request, 'Ad deleted successfully.')
+        return redirect('my_ads')
+
+@login_required
+def ad_detail(request, ad_id):
+    ad = get_object_or_404(Ad, id=ad_id)
+    ad_images = ad.images.all()
+    categories = Category.objects.all()
+    liked_by_user = Like.objects.filter(ad=ad, user=request.user).exists() if request.user.is_authenticated else False
+    receiver = ad.user
+     
+    return render(request, 'ad_detail.html', {
+        'ad': ad,
+        'ad_images': ad_images,
+        'categories': categories,
+        'liked_by_user': liked_by_user,
+        'receiver': receiver, 
+    })
+
+@login_required
+def like_ad(request, ad_id):
+    if request.method == 'POST':
```

Comment:
Instead of checking with IF block, you can use `require_POST` decorator on the view.
https://docs.djangoproject.com/en/5.1/topics/http/decorators/#django.views.decorators.http.require_POST

Reply:
ok